src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -406,7 +408,16 @@ public async Task<JObject> Resolve(ElementAccessExpressionSyntax elementAccess,
                                     indexObject ??= await Resolve(argParm.Identifier.Text, token);
                                     elementIdxStr += indexObject["value"].ToString();
                                 }
-                                // FixMe: indexing with expressions, e.g. x[a + 1]
+                                // indexing with expressions, e.g. x[a + 1]
+                                else
+                                {
+                                    SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(arg + @";", cancellationToken: token);
```

Comment:
IIUC, we can use the `$"{arg};"` directly instead of creating a `SyntaxTree`. Creating that would be useful if we wanted to make changes to the expression.

Reply:
Oh, sorry, `arg` is not a string. It think you should use `arg.ToString()` for both the arguments passed to `EvaluateSimpleExpression`.